src/uct/tcp/tcp_iface.c

```diff
@@ -238,7 +242,24 @@ uct_tcp_iface_is_reachable_v2(const uct_iface_h tl_iface,
 
     /* Later connect() call can still fail if the peer is actually unreachable
      * at UCT/TCP EP creation time */
-    return uct_iface_scope_is_reachable(tl_iface, params);
+    if (!uct_iface_scope_is_reachable(tl_iface, params)) {
+        return 0;
+    }
+
+    status = ucs_ifname_to_index(iface->if_name, &ndev_index);
```

Comment:
1) add documentation to describe the new logic
2) use `uct_iface_fill_info_str_buf` for error msg